apps/sms/views/conversation/test/unit/conversation_test.js

```diff
@@ -2364,6 +2364,61 @@ suite('conversation.js >', function() {
         assert.isTrue(node.classList.contains('read'));
       }).then(done, done);
     });
+
+    test('display late arrival notice when required', function(done) {
+      const SECOND = 1000;
+      const MINUTE = 60 * SECOND;
+      const HOUR = 60 * MINUTE;
+      const DAY = 24 * HOUR;
+      const MONTH = 30 * DAY;
+
+      this.sinon.clock.tick(10 * MONTH);
+      var now = Date.now();
+      var delays = [
+        SECOND,
+        MINUTE - 1,
+        MINUTE,
+        HOUR - 1,
+        HOUR,
+        DAY - 1,
+        DAY,
+        MONTH - 1,
+        MONTH,
+        MONTH * 5
+      ];
+      var dataL10nIds = [
+        null,
+        null,
+        'late-arrival-notice-in-minutes',
+        'late-arrival-notice-in-minutes',
+        'late-arrival-notice-in-hours',
+        'late-arrival-notice-in-hours',
+        'late-arrival-notice-in-days',
+        'late-arrival-notice-in-days',
+        'late-arrival-notice-in-months',
+        'late-arrival-notice-in-months'
+      ];
+
+      Promise.all(
+        delays.map((delay, index) => {
+          var message =MockMessages.sms({
+            delivery: 'received',
+            timestamp: now,
+            sentTimestamp: now - delay
+          });
+
+          return ConversationView.buildMessageDOM(message).then((node) => {
+            var noticeNode = node.querySelector('.late-arrival-notice');
+            if(index < 2){
+              assert.isNull(noticeNode);
+            } else {
+              assert.isNotNull(noticeNode);
+              assert.equal(noticeNode.dataset.l10nId, dataL10nIds[index]);
+            }
+          });
+        })
+      ).then(() => done(), () => done());
```

Comment:
this won't work: you're gobbling all errors with this then :)

i'd propose to generate 1 test for each delay, instead. See https://github.com/mozilla-b2g/gaia/blob/e21a2e5c1fb7bf329553a2ae98a8b9e6bef0028b/apps/sms/views/shared/test/unit/startup_test.js#L120 as an example